src/net_processing.cpp

```diff
@@ -2572,33 +2579,79 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
-                // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
-                assert(recentRejects);
-                recentRejects->insert(tx.GetHash());
-                if (RecursiveDynamicUsage(*ptx) < 100000) {
-                    AddToCompactExtraTransactions(ptx);
+            // If this tx didn't make it in due to feerate, and there is a tx
+            // in the orphan pool -- then maybe that tx is only missing this
+            // one parent.
+            // Try to process the pair as a package.
+            bool added_as_package = false;
+            if (state.GetRejectCode() == REJECT_INSUFFICIENTFEE) {
+                std::list<std::map<uint256, COrphanTx>::iterator> orphans_missing_this_tx;
+                for (size_t i=0; i<tx.vout.size(); ++i) {
+                    auto it = mapOrphanTransactionsByPrev.find(COutPoint(tx.GetHash(), i));
+                    if (it != mapOrphanTransactionsByPrev.end()) {
+                        for (auto orphan_iter : it->second) orphans_missing_this_tx.push_back(orphan_iter);
+                    }
+                }
+                if (!orphans_missing_this_tx.empty()) {
+                    const COrphanTx &orphan_tx = orphans_missing_this_tx.front()->second;
+                    // Pick the first transaction, and process the pair. If it's
+                    // missing other inputs, this will of course fail.
+                    std::list<CTransactionRef> package;
+                    package.push_back(ptx);
+                    package.push_back(orphan_tx.tx);
+                    CValidationState package_state;
+                    if (AcceptPackageToMemoryPool(mempool, state, package, nullptr, &lRemovedTxn, 0 /* nAbsurdFee */, false /* test_accept */)) {
+                        LogPrintf("package accepted!!\n"); // XXX: improve logging
+                        added_as_package = true;
+                        mempool.check(&::ChainstateActive().CoinsTip());
+                        EraseOrphanTx(orphan_tx.tx->GetHash());
+                        for (auto package_tx : package) RelayTransaction(package_tx->GetHash(), *connman);
+                        for (auto package_tx : package) UpdateOrphanWorkSet(*package_tx, pfrom);
+
+                        pfrom->nLastTXTime = GetTime();
+
+                        for (auto package_tx : package) {
+                            LogPrint(BCLog::MEMPOOL, "AcceptToMemoryPool: peer=%d: accepted %s (poolsz %u txn, %u kB)\n",
+                                    pfrom->GetId(),
+                                    package_tx->GetHash().ToString(),
+                                    mempool.size(), mempool.DynamicMemoryUsage() / 1000);
+                        }
+
+                        // Recursively process any orphan transactions that depended on these
+                        ProcessOrphanTx(connman, pfrom->orphan_work_set, lRemovedTxn);
+                    }
```

Comment:
If package wasn't accepted (like a high-fee invalid orphan tx),  and if inputs are not missing, if orphan_tx is invalid we may punish peer, or at least erase it for not being standard or insufficient fee?